lib/Conversion/TorchToLinalg/Utils.cpp

```diff
@@ -116,6 +117,34 @@ Value torch_to_linalg::getOutputDimForConvOps(OpBuilder &b, Location loc,
   else
     division = b.createOrFold<arith::FloorDivSIOp>(loc, dividend, strideInt);
   Value out = b.createOrFold<arith::AddIOp>(loc, division, c1);
+
+  if (ceilMode) {
+    Value outMinusOneTimesStride =
+        b.createOrFold<arith::MulIOp>(loc, division, strideInt);
+    Value inAddLeftPadding = b.createOrFold<arith::AddIOp>(
+        loc, castIndexToInt64(b, loc, in), paddingInt);
+
+    auto reduceOutputDim =
+        b.createOrFold<arith::CmpIOp>(loc, arith::CmpIPredicate::uge,
+                                      outMinusOneTimesStride, inAddLeftPadding);
+
+    // Emit 'then' region of 'scf.if'
+    auto emitThenRegion = [&](OpBuilder &opBuilder, Location loc) {
+      opBuilder.create<scf::YieldOp>(loc, division);
+    };
+
+    // Emit 'else' region of 'scf.if'
+    auto emitElseRegion = [&](OpBuilder &opBuilder, Location loc) {
+      opBuilder.create<scf::YieldOp>(loc, out);
+    };
+
+    // Emit 'scf.if' op
+    auto ifOp = b.create<scf::IfOp>(loc, reduceOutputDim, emitThenRegion,
+                                    emitElseRegion);
+
+    return castIntToIndex(b, loc, ifOp.getResult(0));
```

Comment:
You can also use `arith.Select` op in place of `scf.if`.

Reply:
Fixed.